pkg/webhooks/techpreviewnoupgrade/techpreviewnoupgrade.go

```diff
@@ -103,23 +101,31 @@ func (s *TechPreviewNoUpgradeWebhook) renderFeatureGate(request admissionctl.Req
 }
 
 func (s *TechPreviewNoUpgradeWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
 	featureGate, err := s.renderFeatureGate(request)
 
 	if err != nil {
-		return admissionctl.Errored(http.StatusBadRequest, err)
+		log.Error(err, "Couldn't render a FeatureGate from the incoming request")
+
+		ret = admissionctl.Errored(http.StatusOK, err)
```

Comment:
Why does this need to change from StatusBadRequest to StatusOK?

Reply:
Because the request is not really malformed or bad in itself. The operation is not authorized to be performed. 
I am more open to ideas in this, should we instead use `StatusForbidden`? @NautiluX

Reply:
I'm still not sure what you mean by the operation is not authorized, doesn't that error mean the request couldn't be parsed into a FeatureGate? So wouldn't that be a bad payload rather than a forbidden request?

Reply:
Oh I was thinking in the sense that this response is being generated from the `authorised` method. When this block of code executes, the request is not authorized since the error response is returned. 

But I agree with you now that I realize that this is meant for requests not having a well formed FeatureGate. I will restore it back to `StatusBadRequest`.

Reply:
@NautiluX I have reverted this to `StatusBadRequest` as per discussion above.